modules/foreman_setting_facts.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2018 Matthias M Dellweg (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_setting_facts
+short_description: Gather facts about Foreman Settings
+description:
+  - "Gather facts about Foreman Setting Entities"
+  - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.4"
+author:
+  - "Matthias M Dellweg (@mdellweg) ATIX AG"
+requirements:
+  - nailgun >= 0.29.0
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+  name:
+    description:
+      - Name of the Setting to fetch
+      - If not given, fetch all settings
+    required: false
+'''
+
+EXAMPLES = '''
+- name: "Read a Setting"
+  foreman_setting_facts:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "http_proxy"
+  register: result
+
+- name: "Read all Settings"
+  foreman_setting_facts:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+  register: result
+'''
```

Comment:
Can you add an example of using the registered var ? The `stat` module has examples like this and it makes it easier to use. Maybe one example with `name` and one without would be good.